File: OTCDerivativesCalculatorModule/Project_Cpp/lib_static/FpmlSerialized/GenClass/fpml-shared-5-4/ExternalDocument.cpp
// ExternalDocument.cpp 
#include "ExternalDocument.hpp"
#ifdef ConsolePrint
    #include <iostream>
#endif 

namespace FpmlSerialized {

ExternalDocument::ExternalDocument(TiXmlNode* xmlNode)
: ISerialized(xmlNode)
{
    #ifdef ConsolePrint
        std::string initialtap_ = FileManager::instance().tap_;
        FileManager::instance().tap_.append("   ");
    #endif 
   //mimeTypeNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* mimeTypeNode = xmlNode->FirstChildElement("mimeType");

   if(mimeTypeNode){mimeTypeIsNull_ = false;}
   else{mimeTypeIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- mimeTypeNode , address : " << mimeTypeNode << std::endl;
   #endif
   if(mimeTypeNode)
   {
      if(mimeTypeNode->Attribute("href") || mimeTypeNode->Attribute("id"))
      {
          if(mimeTypeNode->Attribute("id"))
          {
             mimeTypeIDRef_ = mimeTypeNode->Attribute("id");
             mimeType_ = boost::shared_ptr<MimeType>(new MimeType(mimeTypeNode));
             mimeType_->setID(mimeTypeIDRef_);
             IDManager::instance().setID(mimeTypeIDRef_,mimeType_);
          }
          else if(mimeTypeNode->Attribute("href")) { mimeTypeIDRef_ = mimeTypeNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { mimeType_ = boost::shared_ptr<MimeType>(new MimeType(mimeTypeNode));}
   }

   //stringNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* stringNode = xmlNode->FirstChildElement("string");

   if(stringNode){stringIsNull_ = false;}
   else{stringIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- stringNode , address : " << stringNode << std::endl;
   #endif
   if(stringNode)
   {
      if(stringNode->Attribute("href") || stringNode->Attribute("id"))
      {
          if(stringNode->Attribute("id"))
          {
             stringIDRef_ = stringNode->Attribute("id");
             string_ = boost::shared_ptr<XsdTypeString>(new XsdTypeString(stringNode));
             string_->setID(stringIDRef_);
             IDManager::instance().setID(stringIDRef_,string_);
          }
          else if(stringNode->Attribute("href")) { stringIDRef_ = stringNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { string_ = boost::shared_ptr<XsdTypeString>(new XsdTypeString(stringNode));}
   }

   //hexadecimalBinaryNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* hexadecimalBinaryNode = xmlNode->FirstChildElement("hexadecimalBinary");

   if(hexadecimalBinaryNode){hexadecimalBinaryIsNull_ = false;}
   else{hexadecimalBinaryIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- hexadecimalBinaryNode , address : " << hexadecimalBinaryNode << std::endl;
   #endif
   if(hexadecimalBinaryNode)
   {
      if(hexadecimalBinaryNode->Attribute("href") || hexadecimalBinaryNode->Attribute("id"))
      {
          if(hexadecimalBinaryNode->Attribute("id"))
          {
             hexadecimalBinaryIDRef_ = hexadecimalBinaryNode->Attribute("id");
             hexadecimalBinary_ = boost::shared_ptr<XsdTypeHexBinary>(new XsdTypeHexBinary(hexadecimalBinaryNode));
             hexadecimalBinary_->setID(hexadecimalBinaryIDRef_);
             IDManager::instance().setID(hexadecimalBinaryIDRef_,hexadecimalBinary_);
          }
          else if(hexadecimalBinaryNode->Attribute("href")) { hexadecimalBinaryIDRef_ = hexadecimalBinaryNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { hexadecimalBinary_ = boost::shared_ptr<XsdTypeHexBinary>(new XsdTypeHexBinary(hexadecimalBinaryNode));}
   }

   //base64BinaryNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* base64BinaryNode = xmlNode->FirstChildElement("base64Binary");

   if(base64BinaryNode){base64BinaryIsNull_ = false;}
   else{base64BinaryIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- base64BinaryNode , address : " << base64BinaryNode << std::endl;
   #endif
   if(base64BinaryNode)
   {
      if(base64BinaryNode->Attribute("href") || base64BinaryNode->Attribute("id"))
      {
          if(base64BinaryNode->Attribute("id"))
          {
             base64BinaryIDRef_ = base64BinaryNode->Attribute("id");
             base64Binary_ = boost::shared_ptr<XsdTypeBase64Binary>(new XsdTypeBase64Binary(base64BinaryNode));
             base64Binary_->setID(base64BinaryIDRef_);
             IDManager::instance().setID(base64BinaryIDRef_,base64Binary_);
          }
          else if(base64BinaryNode->Attribute("href")) { base64BinaryIDRef_ = base64BinaryNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { base64Binary_ = boost::shared_ptr<XsdTypeBase64Binary>(new XsdTypeBase64Binary(base64BinaryNode));}
   }

   //urlNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* urlNode = xmlNode->FirstChildElement("url");

   if(urlNode){urlIsNull_ = false;}
   else{urlIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- urlNode , address : " << urlNode << std::endl;
   #endif
   if(urlNode)
   {
      if(urlNode->Attribute("href") || urlNode->Attribute("id"))
      {
          if(urlNode->Attribute("id"))
          {
             urlIDRef_ = urlNode->Attribute("id");
             url_ = boost::shared_ptr<XsdTypeAnyURI>(new XsdTypeAnyURI(urlNode));
             url_->setID(urlIDRef_);
             IDManager::instance().setID(urlIDRef_,url_);
          }
          else if(urlNode->Attribute("href")) { urlIDRef_ = urlNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { url_ = boost::shared_ptr<XsdTypeAnyURI>(new XsdTypeAnyURI(urlNode));}
   }

   //attachmentReferenceNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* attachmentReferenceNode = xmlNode->FirstChildElement("attachmentReference");

   if(attachmentReferenceNode){attachmentReferenceIsNull_ = false;}
   else{attachmentReferenceIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- attachmentReferenceNode , address : " << attachmentReferenceNode << std::endl;
   #endif
   if(attachmentReferenceNode)
   {
      if(attachmentReferenceNode->Attribute("href") || attachmentReferenceNode->Attribute("id"))
      {
          if(attachmentReferenceNode->Attribute("id"))
          {
             attachmentReferenceIDRef_ = attachmentReferenceNode->Attribute("id");
             attachmentReference_ = boost::shared_ptr<HTTPAttachmentReference>(new HTTPAttachmentReference(attachmentReferenceNode));
             attachmentReference_->setID(attachmentReferenceIDRef_);
             IDManager::instance().setID(attachmentReferenceIDRef_,attachmentReference_);
          }
          else if(attachmentReferenceNode->Attribute("href")) { attachmentReferenceIDRef_ = attachmentReferenceNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { attachmentReference_ = boost::shared_ptr<HTTPAttachmentReference>(new HTTPAttachmentReference(attachmentReferenceNode));}
   }

    #ifdef ConsolePrint
        FileManager::instance().tap_ = initialtap_;
    #endif 
}
boost::shared_ptr<MimeType> ExternalDocument::getMimeType()
{
   if(!this->mimeTypeIsNull_){
        if(mimeTypeIDRef_ != ""){
             return boost::shared_static_cast<MimeType>(IDManager::instance().getID(mimeTypeIDRef_));
        }else{
             return this->mimeType_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<MimeType>();
   }
}
boost::shared_ptr<XsdTypeString> ExternalDocument::getString()
{
   if(!this->stringIsNull_){
        if(stringIDRef_ != ""){
             return boost::shared_static_cast<XsdTypeString>(IDManager::instance().getID(stringIDRef_));
        }else{
             return this->string_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<XsdTypeString>();
   }
}
boost::shared_ptr<XsdTypeHexBinary> ExternalDocument::getHexadecimalBinary()
{
   if(!this->hexadecimalBinaryIsNull_){
        if(hexadecimalBinaryIDRef_ != ""){
             return boost::shared_static_cast<XsdTypeHexBinary>(IDManager::instance().getID(hexadecimalBinaryIDRef_));
        }else{
             return this->hexadecimalBinary_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<XsdTypeHexBinary>();
   }
}
boost::shared_ptr<XsdTypeBase64Binary> ExternalDocument::getBase64Binary()
{
   if(!this->base64BinaryIsNull_){
        if(base64BinaryIDRef_ != ""){
             return boost::shared_static_cast<XsdTypeBase64Binary>(IDManager::instance().getID(base64BinaryIDRef_));
        }else{
             return this->base64Binary_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<XsdTypeBase64Binary>();
   }
}
boost::shared_ptr<XsdTypeAnyURI> ExternalDocument::getUrl()
{
   if(!this->urlIsNull_){
        if(urlIDRef_ != ""){
             return boost::shared_static_cast<XsdTypeAnyURI>(IDManager::instance().getID(urlIDRef_));
        }else{
             return this->url_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<XsdTypeAnyURI>();
   }
}
boost::shared_ptr<HTTPAttachmentReference> ExternalDocument::getAttachmentReference()
{
   if(!this->attachmentReferenceIsNull_){
        if(attachmentReferenceIDRef_ != ""){
             return boost::shared_static_cast<HTTPAttachmentReference>(IDManager::instance().getID(attachmentReferenceIDRef_));
        }else{
             return this->attachmentReference_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<HTTPAttachmentReference>();
   }
}
}

File: OTCDerivativesCalculatorModule/Project_CSharp/XmlBuilder/XmlSerializedClass/fpml-shared-5-4/ExternalDocument.cs
using System;
using System.Text;
using System.Collections.Generic;
using System.Xml;

namespace FpmlSerializedCSharp
{
    public class ExternalDocument : ISerialized
    {
        public ExternalDocument(XmlNode xmlNode)
        {
            XmlNode mimeTypeNode = xmlNode.SelectSingleNode("mimeType");
            
            if (mimeTypeNode != null)
            {
                if (mimeTypeNode.Attributes["href"] != null || mimeTypeNode.Attributes["id"] != null) 
                {
                    if (mimeTypeNode.Attributes["id"] != null) 
                    {
                        mimeTypeIDRef_ = mimeTypeNode.Attributes["id"].Value;
                        MimeType ob = new MimeType(mimeTypeNode);
                        IDManager.SetID(mimeTypeIDRef_, ob);
                    }
                    else if (mimeTypeNode.Attributes["href"] != null)
                    {
                        mimeTypeIDRef_ = mimeTypeNode.Attributes["href"].Value;
                    }
                    else
                    {
                        mimeType_ = new MimeType(mimeTypeNode);
                    }
                }
                else
                {
                    mimeType_ = new MimeType(mimeTypeNode);
                }
            }
            
        
            XmlNode stringNode = xmlNode.SelectSingleNode("string");
            
            if (stringNode != null)
            {
                if (stringNode.Attributes["href"] != null || stringNode.Attributes["id"] != null) 
                {
                    if (stringNode.Attributes["id"] != null) 
                    {
                        stringIDRef_ = stringNode.Attributes["id"].Value;
                        XsdTypeString ob = new XsdTypeString(stringNode);
                        IDManager.SetID(stringIDRef_, ob);
                    }
                    else if (stringNode.Attributes["href"] != null)
                    {
                        stringIDRef_ = stringNode.Attributes["href"].Value;
                    }
                    else
                    {
                        string_ = new XsdTypeString(stringNode);
                    }
                }
                else
                {
                    string_ = new XsdTypeString(stringNode);
                }
            }
            
        
            XmlNode hexadecimalBinaryNode = xmlNode.SelectSingleNode("hexadecimalBinary");
            
            if (hexadecimalBinaryNode != null)
            {
                if (hexadecimalBinaryNode.Attributes["href"] != null || hexadecimalBinaryNode.Attributes["id"] != null) 
                {
                    if (hexadecimalBinaryNode.Attributes["id"] != null) 
                    {
                        hexadecimalBinaryIDRef_ = hexadecimalBinaryNode.Attributes["id"].Value;
                        XsdTypeHexBinary ob = new XsdTypeHexBinary(hexadecimalBinaryNode);
                        IDManager.SetID(hexadecimalBinaryIDRef_, ob);
                    }
                    else if (hexadecimalBinaryNode.Attributes["href"] != null)
                    {
                        hexadecimalBinaryIDRef_ = hexadecimalBinaryNode.Attributes["href"].Value;
                    }
                    else
                    {
                        hexadecimalBinary_ = new XsdTypeHexBinary(hexadecimalBinaryNode);
                    }
                }
                else
                {
                    hexadecimalBinary_ = new XsdTypeHexBinary(hexadecimalBinaryNode);
                }
            }
            
        
            XmlNode base64BinaryNode = xmlNode.SelectSingleNode("base64Binary");
            
            if (base64BinaryNode != null)
            {
                if (base64BinaryNode.Attributes["href"] != null || base64BinaryNode.Attributes["id"] != null) 
                {
                    if (base64BinaryNode.Attributes["id"] != null) 
                    {
                        base64BinaryIDRef_ = base64BinaryNode.Attributes["id"].Value;
                        XsdTypeBase64Binary ob = new XsdTypeBase64Binary(base64BinaryNode);
                        IDManager.SetID(base64BinaryIDRef_, ob);
                    }
                    else if (base64BinaryNode.Attributes["href"] != null)
                    {
                        base64BinaryIDRef_ = base64BinaryNode.Attributes["href"].Value;
                    }
                    else
                    {
                        base64Binary_ = new XsdTypeBase64Binary(base64BinaryNode);
                    }
                }
                else
                {
                    base64Binary_ = new XsdTypeBase64Binary(base64BinaryNode);
                }
            }
            
        
            XmlNode urlNode = xmlNode.SelectSingleNode("url");
            
            if (urlNode != null)
            {
                if (urlNode.Attributes["href"] != null || urlNode.Attributes["id"] != null) 
                {
                    if (urlNode.Attributes["id"] != null) 
                    {
                        urlIDRef_ = urlNode.Attributes["id"].Value;
                        XsdTypeAnyURI ob = new XsdTypeAnyURI(urlNode);
                        IDManager.SetID(urlIDRef_, ob);
                    }
                    else if (urlNode.Attributes["href"] != null)
                    {
                        urlIDRef_ = urlNode.Attributes["href"].Value;
                    }
                    else
                    {
                        url_ = new XsdTypeAnyURI(urlNode);
                    }
                }
                else
                {
                    url_ = new XsdTypeAnyURI(urlNode);
                }
            }
            
        
            XmlNode attachmentReferenceNode = xmlNode.SelectSingleNode("attachmentReference");
            
            if (attachmentReferenceNode != null)
            {
                if (attachmentReferenceNode.Attributes["href"] != null || attachmentReferenceNode.Attributes["id"] != null) 
                {
                    if (attachmentReferenceNode.Attributes["id"] != null) 
                    {
                        attachmentReferenceIDRef_ = attachmentReferenceNode.Attributes["id"].Value;
                        HTTPAttachmentReference ob = new HTTPAttachmentReference(attachmentReferenceNode);
                        IDManager.SetID(attachmentReferenceIDRef_, ob);
                    }
                    else if (attachmentReferenceNode.Attributes["href"] != null)
                    {
                        attachmentReferenceIDRef_ = attachmentReferenceNode.Attributes["href"].Value;
                    }
                    else
                    {
                        attachmentReference_ = new HTTPAttachmentReference(attachmentReferenceNode);
                    }
                }
                else
                {
                    attachmentReference_ = new HTTPAttachmentReference(attachmentReferenceNode);
                }
            }
            
        
        }
        
    
        #region MimeType_
        private MimeType mimeType_;
        public MimeType MimeType_
        {
            get
            {
                if (this.mimeType_ != null)
                {
                    return this.mimeType_; 
                }
                else if (this.mimeTypeIDRef_ != null)
                {
                    mimeType_ = IDManager.getID(mimeTypeIDRef_) as MimeType;
                    return this.mimeType_; 
                }
                else
                {
                      return this.mimeType_; 
                }
            }
            set
            {
                if (this.mimeType_ != value)
                {
                    this.mimeType_ = value;
                }
            }
        }
        #endregion
        
        public string mimeTypeIDRef_ { get; set; }
        #region String_
        private XsdTypeString string_;
        public XsdTypeString String_
        {
            get
            {
                if (this.string_ != null)
                {
                    return this.string_; 
                }
                else if (this.stringIDRef_ != null)
                {
                    string_ = IDManager.getID(stringIDRef_) as XsdTypeString;
                    return this.string_; 
                }
                else
                {
                      return this.string_; 
                }
            }
            set
            {
                if (this.string_ != value)
                {
                    this.string_ = value;
                }
            }
        }
        #endregion
        
        public string stringIDRef_ { get; set; }
        #region HexadecimalBinary_
        private XsdTypeHexBinary hexadecimalBinary_;
        public XsdTypeHexBinary HexadecimalBinary_
        {
            get
            {
                if (this.hexadecimalBinary_ != null)
                {
                    return this.hexadecimalBinary_; 
                }
                else if (this.hexadecimalBinaryIDRef_ != null)
                {
                    hexadecimalBinary_ = IDManager.getID(hexadecimalBinaryIDRef_) as XsdTypeHexBinary;
                    return this.hexadecimalBinary_; 
                }
                else
                {
                      return this.hexadecimalBinary_; 
                }
            }
            set
            {
                if (this.hexadecimalBinary_ != value)
                {
                    this.hexadecimalBinary_ = value;
                }
            }
        }
        #endregion
        
        public string hexadecimalBinaryIDRef_ { get; set; }
        #region Base64Binary_
        private XsdTypeBase64Binary base64Binary_;
        public XsdTypeBase64Binary Base64Binary_
        {
            get
            {
                if (this.base64Binary_ != null)
                {
                    return this.base64Binary_; 
                }
                else if (this.base64BinaryIDRef_ != null)
                {
                    base64Binary_ = IDManager.getID(base64BinaryIDRef_) as XsdTypeBase64Binary;
                    return this.base64Binary_; 
                }
                else
                {
                      return this.base64Binary_; 
                }
            }
            set
            {
                if (this.base64Binary_ != value)
                {
                    this.base64Binary_ = value;
                }
            }
        }
        #endregion
        
        public string base64BinaryIDRef_ { get; set; }
        #region Url_
        private XsdTypeAnyURI url_;
        public XsdTypeAnyURI Url_
        {
            get
            {
                if (this.url_ != null)
                {
                    return this.url_; 
                }
                else if (this.urlIDRef_ != null)
                {
                    url_ = IDManager.getID(urlIDRef_) as XsdTypeAnyURI;
                    return this.url_; 
                }
                else
                {
                      return this.url_; 
                }
            }
            set
            {
                if (this.url_ != value)
                {
                    this.url_ = value;
                }
            }
        }
        #endregion
        
        public string urlIDRef_ { get; set; }
        #region AttachmentReference_
        private HTTPAttachmentReference attachmentReference_;
        public HTTPAttachmentReference AttachmentReference_
        {
            get
            {
                if (this.attachmentReference_ != null)
                {
                    return this.attachmentReference_; 
                }
                else if (this.attachmentReferenceIDRef_ != null)
                {
                    attachmentReference_ = IDManager.getID(attachmentReferenceIDRef_) as HTTPAttachmentReference;
                    return this.attachmentReference_; 
                }
                else
                {
                      return this.attachmentReference_; 
                }
            }
            set
            {
                if (this.attachmentReference_ != value)
                {
                    this.attachmentReference_ = value;
                }
            }
        }
        #endregion
        
        public string attachmentReferenceIDRef_ { get; set; }
        public string choiceStr_0;
        
    
        
    
    }
    
}

